Simple Slideshow

printfuck opened the pull request:
This is a simple slideshow feature with a setting for the time interval.

I also admittedly changed the icon placement inside the buttons - this is not necessary, but it made the alignment nicer (in my opinion).

https://github.com/photoview/photoview/assets/36680203/280dc527-33c9-49c7-9d92-ff03561911cb




<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Media gallery now separately tracks photos and videos for improved navigation.
  - Parent components can manage video element references for better playback control.
  - Configurable slideshow with keyboard shortcuts, auto-advance, interval and mode controls (photos, videos, all, repeat).
  - New play/pause/repeat/photo/video icons and UI controls.
  - Improved type safety for media elements.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
Added slideshow implementation

jordy2254 commented:
Thanks for your PR. Sounds like a useful feature. I will checkout your changes during the week to review them.

kkovaletp commented:
I like the slideshow feature from functionality point of view. It will be definitely useful for many users.
Sure, we could start with something simple and then enhance it iteratively.
@printfuck , it would be great if you can describe it with some more details and probably attach some screenshots, so we could clearly understand all the functions and limitations: for instance, is it possible to start it for an album or it can run for all the gallery? will it play videos or skip them? etc...

printfuck commented:
> I like the slideshow feature from the functionality point of view. It will be definitely useful for many users. Sure, we could start with something simple and then enhance it iteratively. @printfuck, it would be great if you could describe it with some more details and probably attach some screenshots, so we could clearly understand all the functions and limitations: for instance, is it possible to start it for an album or it can run for all the gallery? will it play videos or skip them? etc...

Thanks for your reply.
It basically just iterates on all images inside of an album from within the presentView; videos won't be started. My focus was entirely on images, but I think it would make sense to play each video once and cycle by default. I think, there would be need for another switch to change video playback behaviour.

Possible options:
 - only Images
 - only Videos, played once
 - both, videos played once

kkovaletp commented:
Thanks for the explanation and the video attachment with the demo!

> there would be need for another switch to change video playback behaviour.
> 
> Possible options:
> 
> * only Images
> * only Videos, played once
> * both, videos played once

I like the idea of having 1 more setting for this. 
However, there is a tricky moment: according to your demo video, the timeout setting is controlled by an on-screen button, shown for the currently opened media. Then the new switch setting needs to be consistent with the timeout setting, it could be a 3-stages-button near the timeout one with:
- picture icon (default)
- video icon
- an icon with both picture and video icons, located as a stack or sequence
corresponding to proposed states. But here we might confuse users, as showing these buttons on screen while a media is shown automatically sets an expectation for a user that the setting is connected to the context of the current media:
- in the case of the currently shown picture user sets the video only, what is expected and actually performed? what if the current album doesn't contain videos - should we check for that in advance and not allow users to change the setting? how might this check impact the performance of starting the view?
- in the opposite case, should we set the 'videos only' value in advance and disable the setting?

While the described scenario looks consistent to me, I'd like to read your and @jordy2254 opinions, and probably start some discussion to make sure that we understand all possible use cases and design the workflow in an optimal way)

printfuck committed:
slideshow modes

printfuck committed:
minor fix

printfuck commented:
I have added the modes Video+Image, Image-only and Video-only. By default Video+Image is selected. The image-only and respectively the video-only modes will simply stop on the current element, if the demanded media type is not present within the album.

The svg images are placeholders. For demonstration purposes, I used images to which would be easily recognizable. If you could point me at a source for images free of copyright, for usage with this project, I'd happily use those.  (The Play and Pause Buttons are done by me so they can be used - just the other three aren't mine)

There is a quirky implementation using the `aux` variable in `PresentMediaOverlay.tsx`, which covers the edge case of only one video in an album on video-only mode.

The Video doesn't demonstrate much, but it displays the look:

https://github.com/photoview/photoview/assets/36680203/719abb0a-2a15-4fdd-aff4-645174d7e5ed

kkovaletp commented:
@printfuck, Thanks for the implementation and your explanation!
I like how it works now)
However, in your video, it is demonstrated that when some bright content is displayed on the background of the control buttons, they are not readable at all. Is it possible to put some lower-transparency background behind the buttons to make sure they are readable even when displayed on top of some bright content?
Or, as an alternative, invert the buttons' colors depending on the background displayed.
It is difficult for me to predict which solution would be better from the UX side and better fit the product design (I guess that is the 1st one), so, I think that you need to play a bit with them (and probably some other possible solutions) to choose the best one.

Regarding the icons sources, I'm not aware of any project-specific icons library. In general, the reuse of some publicly available content is regulated by the content's policies and licenses. Do you know what they are for those icons? If they are compatible with the project's license, then it should be totally fine to use them here. If not, then you need to search for some alternative icons, compatible with our license (or manually recreate them)

kkovaletp commented:
also, please fix the failed tests and double-check the coverage of your code

printfuck committed:
proper types

printfuck committed:
Working and messy

printfuck committed:
all bugs fixed, tests should also build now

kkovaletp commented:
@printfuck, thanks for the new commits!

@jordy2254, please review it and provide your thoughts

marc7s committed:
Merge branch 'master' into slideshow

marc7s committed:
Slideshow fixes

marc7s commented:
@printfuck Took a look at this PR since this is a feature I really think we are missing. Great work on this!

However, checking out the code I noticed some compiler errors that I started to fix. Before I continue looking into this, I would like to check with you if you are open for further development on this? I can assist you even though my React experience is not overly extensive.

I checked out your branch to my fork, merged with the current `master`, and made some changes to the code. The changes are to fix compiler errors mainly. Also some typing, and a function separation for the next slide logic. I have not yet looked at the tests so they might have errors or may fail due to some typing changes. You can see my changes [in this branch](https://github.com/marc7s/photoview/tree/slideshow). The changes to your code are currently done as a single commit, [and can be viewed here](https://github.com/marc7s/photoview/commit/0ce8aa694836e268a2d529252a241014eda2900e)

If you have time to look at the changes you could maybe integrate them in this PR? I think the slideshow feature is a great addition and would like it merged so I am trying to help on this PR.

We also need to find icons that we can use, but that is a drop in replacement.

printfuck committed:
minor fix for tests to finish

marc7s commented:
@printfuck Thanks for the additions! Would you mind taking a look at the comments by the bot?

As for the images, here are some examples I have found:
# ionicons
License: MIT
[Link](https://ionic.io/ionicons)

[Photo mode](https://unpkg.com/ionicons@7.1.0/dist/svg/image-outline.svg)
[Video mode](https://unpkg.com/ionicons@7.1.0/dist/svg/videocam-outline.svg)
Photo+video mode: Did not find a perfect fit, but perhaps [this?](https://unpkg.com/ionicons@7.1.0/dist/svg/images-outline.svg)
[Repeat mode](https://unpkg.com/ionicons@7.1.0/dist/svg/repeat-outline.svg)

# Phosphoricons
License: MIT
[Link](https://phosphoricons.com/)

Photo mode: `image`
Video mode: `video`
Photo+video mode: Did not find a perfect fit, but perhaps `images`?
Repeat mode: `repeat-once`

For additional icons, React Icons [provides a good table of sources](https://github.com/react-icons/react-icons?tab=readme-ov-file#icons), including their licenses.

kkovaletp committed:
Bump several GO dependencies to the latest versions (#1182)

Co-authored-by: Konstantin Koval

kkovaletp committed:
Bump `gqlgen` to 0.17.70 (#1184)

Co-authored-by: Konstantin Koval

kkovaletp committed:
Bump several deps to the latest versions (#1189)

Co-authored-by: Konstantin Koval

kkovaletp committed:
Bump `gqlgen` and `gqlparser` versions in API (#1191)

* Bump `gqlgen` and `gqlparser` versions in API

* new gqlgen version

---------

Co-authored-by: Konstantin Koval

PeterDaveHello committed:
Add a basic zh-TW Traditional Chinese translation (#1187)

googollee committed:
Add more real media files to `scanner_test` (#1195)

* Add real_media to scanner tests.

* Fix tests.

* Remove old test files.

* Fix with reviews.

kkovaletp committed:
Fix the incorrect thumbnails for albums issue (#1198)

* The fix with test

* Fix test's foreign key issue

* 2 more test cases

* Fix the empty album test expectation

* Return Nil for no thumbnail case instead of an empty album, as expected by frontend. More tests for thumbnail priority choice

* Fix failed tests

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
More granular `media-cache` folder permissions (#1199)

* Add a preparation service to assign the correct permissions to the media cache folder

* Better docker capabilities with less privileges

* Expand the caps list

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Bump GQLgen, GQLparser and GORM versions in API (#1197)

* Bump GQLparser and GORM versions in API

* Bump GQLgen as well

* Revert GORM update

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Handle photo not found case in API routes (#1201)

* Handle media not found issue

* temporary run workflows for master pushes only; tests

* revert temp workflow changes

---------

Co-authored-by: Konstantin Koval

googollee committed:
Remove Thumbnail Preferences. (#1200)

* Remove thumbnail preferences. #1089

* Remove thumbnail preferences test.

* Remove thumbnail preferences from backend.

* Fix database.go

* Update generated TS files.

* Reorder the migrations.

kkovaletp committed:
Bump several API dependencies to their latest versions (#1203)

Co-authored-by: Konstantin Koval

googollee committed:
Drop linux/arm/v7

googollee committed:
Use `magick` generating thumbnails.

Fix #1204

* Add bad medias to scanner_test.

* Fix the script.

* Add GenerateThumbnail function.

* Add identify dimension

* Use new API. Cover json condition.

* Use magick to generate thumbnails.

* Update go mod.

* Remove dependency of executable_worker -> test_utils

* Reduce dependency of magick.

* Use magick to get dimension.

* Remove almost all code around `image` package.

* Fix thumbnailing with multi-frame gif.

* Remove the struct in wrapper.

* Fix typo.

* Build images.

* Update logs

* Add jpeg decoder.

* Remove code dependency.

* Add log to confirm scan in parallel.

* Update log

* Add reformat image perf tests.

* Rollback the build workflow.

googollee committed:
Include MagicWand library in the `dependency` image. (#1209)

* Include MagicWand library.

* Remove duplication.

* Limit dynamic libraries in the release build.

* Use cp -a instead of -r.

* Use bash instead of sh.

googollee committed:
Enable shared build of ImageMagick.

#1211

* Add benchmark magickwand.

* Enable shared magick.

* Remove MagickWand test because it can't be compiled with current dependency image.

* Enable formats explicitly.

kkovaletp committed:
Bump `gorm` and its `postgres` driver to the latest versions (#1215)

Co-authored-by: Konstantin Koval

googollee committed:
Use Magickwand to handle images. (#1212)

* Add MagickWand perf test.

* Use MagickWand.

* Calculate the right thumbnail dimension.

* Remove magick bin file.

* Add building dependencies.

* Build images.

* Add --no-install-recommends

* Remove unused field.

* Fix typo

* Use uint in the worker.

* Add guard

* Always update apt.

* Fix EncodeJpeg() with the quality

* Rename the local error with a better name.

* Add worker.Terminate()

* Fix typo

* Rebase with master and rollback the build workflow.

marc7s committed:
Add multiple face group merge support (#1162)

* Migrate multiple merge faces code from merge-multiple branch

* Revert translations

* Remove unnecessary node

* Add aria labels

* Code review changes

* Add frontend test for MergeFaceGroupsModal

googollee committed:
Update log for passing attrs with context. (#1216)

* Update log for passing attrs with context.

* Check type-casting

* add tests.

* Use t.Cleanup() to restore the global value.

* Fix error.

* Update tests.

* Add comments.

* Remove cover.cov

* Update doc.

* Update git ignore.

kkovaletp committed:
Refactor the Periodic Scanner module (#1210)

* Initial fix of a race condition

* better logging

* some refactoring and graceful shutdown

* fix more potential race condition places; a better app shutdown

* add test and fix some issues

* some housekeeping

* more descriptive asserts

* Address review suggestions

* Better server shutdown logic + better message about the .env file not found

* Create context inside the shutdown goroutine; removed unnecessary shutdown from the main function

* Move HTTP server to the main process and remove unnecessary code

* log shutdown error if any

* Get back to the local `scanner` variable

* Protecting scanner with mutex

* Adding the mutex to the shutdown function

* Addressing the race condition at the init stage

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Bump API deps from last 2 weeks (#1218)

Co-authored-by: Konstantin Koval

kkovaletp committed:
Refactor API video route (#1202)

* Refactored code; temp disable workflows on push

* remove printf leftovers

* A better sanitizing code

* extend auth routes tests

* more tests, fix 2 bugs in token processing

* more refactoring and fixes

* optimize video mediaURLs DB query and make testCachePath in utils thread-safe

* fix always passing test

* and more refactoring; 1 more test, but cleanup still has to be fixed

* more debug code in the test

* fix the test

* Final commit

* Adding a defer function just in case of panic

* Address some of review comments

* Protect the shared var by mutex; implement correct context-aware scanning

* Better error; better initial func wrap/capture; better log messages in `photos.go`;

* Addressing a few more review comments

* Order the query results; a better error type catch; setting the MP4 content type header explicitly

* try to fix the cancelation detection condition

* Extract a function and call it by name

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Bump API deps to versions, released this week (#1220)

Co-authored-by: Konstantin Koval

kkovaletp committed:
Refactor Docker build workflows (#1221)

* Initial implementation

* Multiple fixes and improvemens

* Better `curl` retries timing

* Remove unused `TARGETOS` and `TARGETVARIANT` vars; add the version fetch fallback code to the build scripts

* Move the fallback code before the var usage

* Use double braces to make the code compatible with the `set -u`

---------

Co-authored-by: Konstantin Koval

googollee committed:
Drop tables instead of truncating tables. (#1224)

kkovaletp committed:
Fix the local build of the dependencies image, including the debug image instructions (#1229)

* Fix the local build of the dependencies image, including the debug image instructions

* Make the deps image easily runnable

* improve readme

* Better caching strategy

* make dependency from IM to libs build stages

* fix missing folder error

* Hopefully, final solution

---------

Co-authored-by: Konstantin Koval

googollee committed:
Change default thumbnail of album. (#1225)

dependabot[bot] committed:
Bump github.com/go-viper/mapstructure/v2 (#1230)

Bumps the go_modules group with 1 update in the /api directory: [github.com/go-viper/mapstructure/v2](https://github.com/go-viper/mapstructure).


Updates `github.com/go-viper/mapstructure/v2` from 2.2.1 to 2.3.0
- [Release notes](https://github.com/go-viper/mapstructure/releases)
- [Changelog](https://github.com/go-viper/mapstructure/blob/main/CHANGELOG.md)
- [Commits](https://github.com/go-viper/mapstructure/compare/v2.2.1...v2.3.0)

---
updated-dependencies:
- dependency-name: github.com/go-viper/mapstructure/v2
  dependency-version: 2.3.0
  dependency-type: indirect
  dependency-group: go_modules
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kkovaletp committed:
Bump GQLparser and GQLgen deps in API (#1234)

Co-authored-by: Konstantin Koval

googollee committed:
Ensure features are enabled with imagemagick. (#1233)

googollee committed:
Make TaskContext as context.Context. (#1232)

kkovaletp commented:
@vl-ivanov, please review this PR.

kkovaletp committed:
Fix GitHub commit passing, UI build and cache for different purposes (#1239)

* Fix GitHub commit passing, UI build and cache for different purposes

* a logging for transparency and easy debug

* Fix var expanding with quotes and add intermediate verification commands

* split verification commands

* fix joining commands

* Ecplicitly pull latest deps image

* properly set arg default values and try to fix preparation step

* another try to build the image with the correct deps tag

* Revert the dynamic deps tag discovery, as it doesn't work and the actual `latest` image is pulled implicitly anyway

* Final optimizations

* "NoCommit" placeholder, indent fix, var rename

---------

Co-authored-by: Konstantin Koval

googollee committed:
Fix scanner test. (#1241)

kkovaletp committed:
Fix failed build for `v2.4.0` version (#1244)

* Fix failed build for `v2.4.0` version

* fix the typo

---------

Co-authored-by: Konstantin Koval

googollee committed:
Non-fork face detection, avoiding a race condition. (#1243)

* Non-fork face detection, avoiding a race condition.

* Update scanner test.

kkovaletp committed:
Auto-orient image during processing (#1242)

* Auto-orient image during processing

* Detect if thumb was rotated 90 degrees and switch dimentions

* Consistently auto-rotate images

* fix incorrect path for the original photo dimensions reading

* Swap the dimensions if needed and call the `wand.ThumbnailImage()` unconditionally

* revert unnecessary line split

---------

Co-authored-by: Konstantin Koval

vl-ivanov commented:
Overall, the code looks fine, but there are outstanding issues.
I am not sure if the original author of PR @printfuck is going to apply the missing suggested changes by @coderabbit
since the code is pretty old. 
I will fork his branch, update it and apply the missing changes.

vl-ivanov commented:
@kkovaletp basically, the original PR should be updated with
- the changes from this commit https://github.com/marc7s/photoview/commit/0ce8aa694836e268a2d529252a241014eda2900e done by @marc7s 
- the latest suggestions by Mr. Bunny :)
- adding black border to the SVG icons so they are visible on both light and dark background
- Writing unit tests

You are suggesting to not spend time on this. Why? I mean, are there any other PRs with higher priority? This one seems to be valuable feature IMO.

kkovaletp committed:
Bump several API deps (#1247)

Co-authored-by: Konstantin Koval

googollee committed:
Fix the coverage for codecov. (#1251)

googollee committed:
Add blurhash task, instead of a global blurhash generating. (#1250)

kkovaletp committed:
Small Dockerfile improvements (#1246)

* Small Dockerfile improvements + a cleaner image rotation condition

* quoting

* Revert the change in the `magickwand.go`

---------

Co-authored-by: Konstantin Koval

googollee committed:
Add orientation tests. (#1249)

googollee committed:
Set AutoOrient when reading files. (#1254)

kkovaletp committed:
Add missing `os.exit()` wrapper to the test call. (#1257)

Co-authored-by: Konstantin Koval

googollee committed:
Improve `*TestMain` to handle the exit code. (#1258)

KalininN committed:
Fix: Exif parsers should accept longtitude up to 180 degrees (#1260)

kkovaletp committed:
Bump GORM API dependency to a minor version (#1263)

Co-authored-by: Konstantin Koval

googollee committed:
Add jpeg-xl example file. (#1266)

kkovaletp committed:
Bump GQLgen to the latest version (#1267)

Co-authored-by: Konstantin Koval

tyler-hagen committed:
Update macos local development dependencies (#1268)

* Add needed homebrew formulae and corresponding compiler settings for macos to run api

* coderabbitai feedback

* Don't set CPLUS_INCLUDE_PATH de-novo

kkovaletp committed:
Bump several deps (#1274)

Co-authored-by: Konstantin Koval <kkb@ukr.net>

dependabot[bot] committed:
Bump actions/checkout from 4 to 5 in /.github/workflows (#1277)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang from 1.24-bookworm to 1.25-bookworm (#1278)

* Bump golang from 1.24-bookworm to 1.25-bookworm

Bumps golang from 1.24-bookworm to 1.25-bookworm.

---
updated-dependencies:
- dependency-name: golang
  dependency-version: 1.25-bookworm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump the GO version in the go.mod

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Konstantin Koval <kkb@ukr.net>

kkovaletp committed:
A few small fixes for bugs, detected during the merge upstream changes to my forked repo (#1270)

* Migrate GolangCI config to the latest version; fix nll-pointer exception in the external EXIF parser and order of GPS coords in the internal one; updated links in Readme and added missing LibHeif exports

* Improve GPS error handling

---------

Co-authored-by: Konstantin Koval

googollee committed:
Avoid all os.Exit and log.Fatal (#1279)

* Avoid all os.Exit and log.Fatal

* Exit test with 1 by default, to catch panic errors.

* Fix log messages.

kkovaletp committed:
Monitor both Dockerfiles; Monitor compose files (#1280)

Co-authored-by: Konstantin Koval

googollee committed:
Update dependencies with Debian trixie (#1285)

* Update dep to trixie.

* Update image tag.

googollee committed:
Update API with Debian trixie (#1286)

Fix #1273.

- Cleanup `libatlas-base-dev`

  `ATLAS` is an optimized BLAS implementation. Today, the project looks dead. The last stable release (3.10.3)
happened in 2016. The last development release (3.11.41, not packaged) was in 2018. The git repository has seen no commit since 2019. Debian decided to [remove all `libatlas` packages](https://lists.debian.org/debian-science/2023/07/msg00010.html), including `libatlas-base-dev` from `trixie`.

  `libatlas-base-dev` provides a unique `cblas` package with `pkg-config`. Debian provides [`blas`](https://lists.debian.org/debian-science/2023/07/msg00012.html) to cover all the feature in `cblas`. `go-face` already depends on [`blas`](https://github.com/Kagami/go-face/blob/0c14797b4d0e0a8ff4e183ddbd93cab35a42f0f8/face.go#L4), and it's safe to remove `cblas`.

  Maybe we should fork `go-face` and put those changes in the foked repo.

- Renaming `exiftool`

  In `trixie`, `exiftool` package is renamed as `libimage-exiftool-perl`.

kkovaletp committed:
Bump PostgreSQL image to version 17; add the auto-migrate preparation service (#1281)

* Bump PostgreSQL image to version 17; add the auto-migrate preparation service

* Address bot's comments

* Fix many warnings in the Readme; added better explanations

* Address bot's comments

* And a few more comments addressed

* 1 more small wording fix

* Better clarify the HW acceleration options

* add link to the pgsql upgrade guide

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Set the new PostgreSQL version in dev-compose and API tests (#1289)

Co-authored-by: Konstantin Koval

tyler-hagen committed:
Add scrollbar to people sidebar section (#1276)

* Add scrollbar to people sidebar section

* CR feedback

* Use grid layout to lay 3 images per row

* Center face image over face details text

* Remove margin: auto from circleimagewrapper and align everything with a flexbox on each list element

* Flip menu start of every new row

kkovaletp committed:
Bump `testify` version in API (#1292)

Co-authored-by: Konstantin Koval

googollee committed:
Remove internal exif  (#1293)

* Remove internal exif library.

* Cleanup go.mod.

kkovaletp committed:
Fix filter assignment; `WHERE` statements; added ordering to make the most appropriate URLs go 1st (#1294)

Co-authored-by: Konstantin Koval

googollee committed:
Move `exif` under `externaltools`. (#1297)

* Move exif.

* Better name.

* Add tests and fix codes.

* Rename cleanup. Remove context because it doesn't use.

* Update comments.

* Protect the global var with sync.Once.

* Avoid exit

* Fix the pointer.

* Apply recommends.

* Fix parsing date.

* Fix shadow name.

* Add log when exif parsing has parse failures.

* Add error context.

* Fix advices.

kkovaletp committed:
Bump Gorm and Testify versions (#1302)

Co-authored-by: Konstantin Koval

kkovaletp committed:
Minor improvements for the `/api/utils/utils.go` (#1306)

* Minor improvements for the `/api/utils/utils.go`

* addressing suggestions

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Use correct platform names for QEMU and Docker (#1304)

Co-authored-by: Konstantin Koval

dependabot[bot] committed:
Bump hadolint/hadolint-action from 3.1.0 to 3.2.0 in /.github/workflows (#1309)

Bumps [hadolint/hadolint-action](https://github.com/hadolint/hadolint-action) from 3.1.0 to 3.2.0.
- [Release notes](https://github.com/hadolint/hadolint-action/releases)
- [Changelog](https://github.com/hadolint/hadolint-action/blob/master/.releaserc)
- [Commits](https://github.com/hadolint/hadolint-action/compare/v3.1.0...v3.2.0)

---
updated-dependencies:
- dependency-name: hadolint/hadolint-action
  dependency-version: 3.2.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kkovaletp committed:
Build development UI code when needed (#1305)

* Build development UI code when needed

* add the `--sourcemap --minify=false` options to the dev build

* Add the `--no-audit --no-fund` to NPM CI

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Remove the dependency on the serving UI from the API URI base path logic (#1311)

* Remove the dependency on the serving UI from the API URI base path logic

* address comments

* reuse the same logic consistently

* Validate IP and port inputs; construct the URL directly from the validated parts; consistent URI validations

* Address review comments

---------

Co-authored-by: Konstantin Koval

kkovaletp committed:
Remove Husky dependency and pre-commit hooks from the project (#1312)

* Remove Husky dependency and pre-commit hooks from the project

* Add 2 more scripts to the package.json

---------

Co-authored-by: Konstantin Koval <kkb@ukr.net>

kkovaletp committed:
Bump several API deps to the latest versions (#1316)

Co-authored-by: Konstantin Koval

printfuck committed:
New Images and minor refactor

printfuck committed:
Merge branch 'slideshow'

printfuck committed:
fix video icon + fix hidden properties

printfuck committed:
fix hide property on Image icon
